baby-gru/src/components/validation-tools/MoorhenCarbohydrateValidation.tsx

```diff
@@ -0,0 +1,182 @@
+import { moorhen } from "../../types/moorhen";
+import { useDispatch, useSelector } from 'react-redux';
+import { MoorhenLigandCard } from "../card/MoorhenLigandCard";
+import { Fragment, useCallback, useEffect, useRef, useState } from "react";
+import { Col, Row, Form, Button } from 'react-bootstrap'
+import { MoorhenMapSelect } from '../select/MoorhenMapSelect'
+import { MoorhenMoleculeSelect } from '../select/MoorhenMoleculeSelect'
+import {PrivateerResultsEntry} from "../../types/privateer"
+import {setHoveredAtom} from "../../moorhen"
+interface Props extends moorhen.CollectedProps {
+    dropdownId: number;
+    accordionDropdownId: number;
+    setAccordionDropdownId: React.Dispatch<React.SetStateAction<number>>;
+    sideBarWidth: number;
+    showSideBar: boolean;
+}
+
+function PrivateerValidation(props: {data: PrivateerResultsEntry[], selectedModel: any, clickCallback: (e) => void}) {
+
+    const molecules = useSelector((state: any) => state.molecules);
+    async function handleClick(e) {
+        const newCenterString =
+            e.target.dataset.chainid +
+            '/' +
+            e.target.dataset.seqnum +
+            '(' +
+            e.target.dataset.resname +
+            ')';
+
+        props.clickCallback(newCenterString);
+        const selectedMolecule = molecules.find(
+            (molecule) => molecule.molNo === props.selectedModel
+        );
+        const _center = await selectedMolecule.centreOn(newCenterString);
+    }
+
+    const sugarRef = useCallback(
+        (node: HTMLElement | null) => {
+            if (node !== null) {
+                node.querySelector('svg').style.display = 'block';
+                node.querySelector('svg').style.margin = 'auto';
+
+                const useList = node.querySelectorAll('use');
+
+                for (let i = 0; i < useList.length; i++) {
+                    useList[i].addEventListener('click', handleClick);
+                    useList[i].addEventListener('mousedown', (e) => {
+                        e.stopPropagation();
+                    });
+                    useList[i].addEventListener('touchstart', (e) => {
+                        e.stopPropagation();
+                    });
+                }
```

Comment:
I'm not sure I follow this but do we need to remove this evt listeners at some point or do they simply get removed when the svg disappear from the DOM?